src/async_monitored_item.rs

```diff
@@ -262,8 +262,16 @@ impl<K: MonitoredItemKind> MonitoredItemBuilder<K> {
             let request = ua::DeleteMonitoredItemsRequest::init()
                 .with_subscription_id(subscription_id)
                 .with_monitored_item_ids(&monitored_item_ids);
-            // This request is processed asynchronously. Errors are logged asynchronously too.
-            crate::delete_monitored_items::call(client, &request);
+            // Await the response to ensure that all previously created monitored items
```

Comment:
The behavior has change for this unexpected and unlikely case.

In an async context we could and should await the response and log any failures here.

Reply:
We could safely take out time here. It is an unexpected error case and the logs should be precise if this ever happens. No need to rush.

Reply:
In this narrow context we want to make sure that the caller can safely assume that nothing has changed on the server if the request returned with an error. Serializing the rollback call (request/response) within the asynchronous call sequence/span reflects this intent.

Other requests are not affected and could safely be interleaved to reduce latencies.